src/generate/convert/mod.rs

```diff
@@ -271,6 +274,32 @@ pub fn convert_node(ast: &ASTTy, imp: &mut Imports, state: &State, ctx: &Context
                 op: CoreOp::Assign,
             },
         }
+    } else if last_ret {
+        match core {
+            Core::Block { ref statements } => if let Some(last) = statements.last() {
+                match last {
+                    Core::Return { .. } => core.clone(),
+                    _ => {
+                        let mut new_stmts = statements.clone();
+                        let last = if let Some(last) = new_stmts.pop() {
+                            match last {
+                                Core::IfElse { .. } | Core::Raise { .. } => last.clone(), // Ignore if-else and raise
+                                _ => Core::Return { expr: Box::from(last.clone()) }
+                            }
+                        } else {
+                            Core::Return { expr: Box::from(Core::None) }
+                        };
+
+                        new_stmts.push(last);
+                        Core::Block { statements: new_stmts }
+                    }
+                }
+            } else {
+                core.clone()
+            }
+            Core::Return { .. } | Core::IfElse { .. } | Core::Raise { .. } => core,
```

Comment:
Are we absolutely sure these are the only constructs?
Best to make this a named function as well for readability.
I'm guessing that stuff like `for loop`s and `while loop`s should not have a return (Even if they won't pass the check stage if a function ends with those).